tests/e2e/testBatteries.spec.ts

```diff
@@ -0,0 +1,159 @@
+ *      {
+ *        $: "selector", // optional, "button" will be used if not specified
+ *        innerText: "text content of the button", // optional
+ *        waitNavigator: true, // wait for page to be idle after clicking
+ *        scrollDown: true // scroll the page down before clicking (this is used eg. for the cookies notice)
+ *      }
+ */
+async function clickButton(page, selectionData) {
+  const selector =
+    typeof selectionData == "object" && selectionData.$
+      ? selectionData.$
+      : "button";
+  const start = Date.now();
+  return new Promise<void>((resolve) => {
+    const findAndClickButton = async () => {
+      const buttons = await page.$$(selector);
+      for (const button of buttons) {
+        const buttonText = await page.evaluate((el) => el.textContent, button);
+        if (
+          buttonText &&
+          selectionData.innerText &&
+          buttonText.indexOf(selectionData.innerText) == -1
+        )
+          continue;
+        if (selectionData.scrollDown)
+          await page.evaluate(() => window.scrollBy(0, window.innerHeight));
+        await delay(100);
+        await button.click();
+        if (selectionData.waitNavigator) {
+          await page.waitForNavigation({ waitUntil: "networkidle2" });
+        } else {
+          await delay(100);
+        }
+        resolve();
+        return;
+      }
+      console.info("Looking for button ...", selectionData);
+      if (Date.now() - start > 1 * 60 * 1000)
+        resolve(); // timeout after 1 minute
+      else setTimeout(() => findAndClickButton(), 1000);
+    };
+    setTimeout(() => findAndClickButton(), 1000);
+  });
+}
+
+/**
+ * Test an unit
+ * @param page page object
+ * @param unit name of the unit to test
+ */
+async function testUnit(page, unit) {
+  const unitData = Units[unit];
+  if (!unitData) throw `Unit ${unit} not configured`;
+  console.info("Loading page", unitData.url);
+  await page.goto(unitData.url, {
+    waitUntil: "networkidle2",
+  });
+  await delay(1000);
+  if (unitData.cookieButtonSelector)
+    await clickButton(page, unitData.cookieButtonSelector);
+
+  const lightningData = ((await getLightningData(page)) ?? [])[0];
+  console.info("Received lightning data", lightningData);
+  if (!lightningData) throw `Lightning data not found for unit ${unit} :(`;
+  for (const k in unitData.output) {
+    const v = unitData.output[k];
+    if (typeof v === "object") {
+      if (v.startsWith) {
+        if (!lightningData[k].startsWith(v.startsWith))
+          throw `Invalid ${k}. Got ${lightningData[k]} but ${v.startsWith}... was expected`;
+      }
+    } else {
+      if (lightningData[k] != v)
+        throw `Invalid ${k}. Got ${lightningData[k]} but ${v} was expected`;
+    }
+  }
+}
+
+/**
+ * Retrieve the lightning data of the page by querying the extension
+ * @param page page object to query
+ * @returns the lightning data
+ */
+const getLightningData = async (page): Promise<void> => {
```

Comment:
My `void` return type is wrong. Maybe @dylancom has an idea what this should be.

Reply:
Oh, right... it should probably be 'any'

Reply:
Yeah, it was `any` before. Would like to avoid `any`. But if we don't have a choice we can use it, comment it and turn of linting for this line.

Reply:
Mh, the actual type should be Battery from import {Battery} from '~/types', but this file is not in the test source path. We could redefine a type that is a subset of battery but that will add extra maintenance cost, leaving it as any is not too bad imo